src/main/java/com/elytradev/concrete/recipe/FluidIngredient.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2016-2017:
+ * 	William Thompson (unascribed),
+ * 	Isaac Ellingson (Falkreon),
+ * 	Jamie Mansfield (jamierocks),
+ * 	and contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+ * of the Software, and to permit persons to whom the Software is furnished to do
+ * so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.elytradev.concrete.recipe;
+
+import com.elytradev.concrete.recipe.impl.FluidStackIngredient;
+
+import net.minecraftforge.fluids.Fluid;
+import net.minecraftforge.fluids.FluidStack;
+
+public abstract class FluidIngredient implements IIngredient<FluidStack> {
+
+	@Override
+	public String getCategory() {
+		return "fluid";
+	}
+
+	/** Creates a new IIngredient matching the provided FluidStack's fluid and NBT, and having any amount as long as
```

Comment:
Proper javadocs.

Reply:
Please clarify. Do we want param/return annotations? When I was putting these together I felt that I couldn't add any additional clarity by adding them, but welcome your thoughts.

Reply:
Idk if that's what @jamierocks was talking about but having param and return docs even if they're slightly redundant is nice.

Reply:
Idk if that's what @jamierocks was talking about but having param and return docs even if they're slightly redundant is nice.